src/Explorer/Tree/Database.ts

```diff
@@ -388,4 +396,39 @@ export default class Database implements ViewModels.Database {
   private _getOfferForDatabase(offers: DataModels.Offer[], database: DataModels.Database): DataModels.Offer {
     return _.find(offers, (offer: DataModels.Offer) => offer.resource === database._self);
   }
+
+  private addSchema(collection: DataModels.Collection): void {
+    if (collection.analyticalStorageTtl == undefined || !this.container.isSchemaEnabled()) {
+      return;
+    }
+
+    collection.requestSchema = () => {
+      this.junoClient.requestSchema({
+        id: null,
+        subscriptionId: userContext.subscriptionId,
+        resourceGroup: userContext.resourceGroup,
+        accountName: userContext.databaseAccount.name,
+        resource: `dbs/${this.id}/colls/${collection.id}`,
+        status: "new"
+      });
+      const checkForSchema = setInterval(async () => {
+        const response: IJunoResponse<DataModels.ISchema> = await this.junoClient.getSchema(
+          userContext.databaseAccount.name,
+          this.id(),
+          collection.id
+        );
+
+        if (response.status >= 404) {
+          clearInterval(checkForSchema);
+        }
+
+        if (response.data != null) {

```

Comment:
Is there a particular reason to use `!= null` rather than `!== null`?

Reply:
nope, c# muscle memory.